### PR TITLE
Refactored transmissions library

### PR DESCRIPTION
**Pull request by aydrian**
#125 
Removed need for `transmissionBody` object on `send()`
Removed `toApiFormat`, functions expect parameters to match api documentation
Refactoring for better error handling
Updated tests for Promises

### CODE REVIEW

File: test/spec/transmissions.spec.js [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
   , sinon = require('sinon')
-  , sinonChai = require('sinon-chai')
   /* global -Promise */
   , Promise = require('../../lib/Promise');
 
-chai.use(sinonChai);
+require('sinon-as-promised');
+
+chai.use(require('sinon-chai'));
+chai.use(require('chai-as-promised'));
```
Comment (rajumsys) [inline test/spec/transmissions.spec.js]: it looks like all tests are using `done` style test. so why you need `chai-as-promised`?

Comment (aydrian) [inline test/spec/transmissions.spec.js]: There are tests using it. I probably need to move them all over to `chai-as-promised`. But I'm not sure.

File: README.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ npm install sparkpost
     * `options` - [see request modules options](https://github.com/mikeal/request#requestoptions-callback)
     * `options.uri` - can either be a full url or a path that is appended to `options.origin` used at initialization ([url.resolve](http://nodejs.org/api/url.html#url_url_resolve_from_to))
     * `options.debug` - setting to `true` includes full response from request client for debugging purposes
-    * `callback` - executed after task is completed. **required**
+    * `callback` - executed after task is completed if provided
```
Comment (jasonrhodes) [inline README.md]: Can we add a bit to these sections that specify what the method returns (ie a promise) to help clarify this?

Comment (aydrian) [inline README.md]: Yeah. I think I meant to but got hung up on where to put it. Do you have a good example or suggestion?

**Pull request merged**